mathesar_ui/src/pages/database-connection/AddDatabaseConnection.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+  import AppendBreadcrumb from '@mathesar/components/breadcrumb/AppendBreadcrumb.svelte';
+  import { iconAddNew } from '@mathesar/icons';
+  import {
+    DATABASE_CONNECTION_ADD_URL,
+    getDatabasePageUrl,
+  } from '@mathesar/routes/urls';
+  import { toast } from '@mathesar/stores/toast';
+  import type { SuccessfullyConnectedDatabase } from '@mathesar/AppTypes';
+  import { router } from 'tinro';
+  import { reloadDatabases } from '@mathesar/stores/databases';
+  import { reflectApi } from '@mathesar/api/reflect';
+  import FormBox from '../admin-users/FormBox.svelte';
+  import DatabaseConnectionForm from './DatabaseConnectionForm.svelte';
+
+  async function handleSuccess(
+    event: CustomEvent<SuccessfullyConnectedDatabase>,
+  ) {
+    const database = event.detail;
+    toast.success(`${database.name} connected successfully!`);
+    await reflectApi.reflect();
+    await reloadDatabases();
+    router.goto(getDatabasePageUrl(database.name));
```

Comment:
We need to show some loading indication here. These requests might take considerable time, and we're waiting on them before moving to the db page.